src/app/application.cpp

```diff
@@ -186,7 +186,21 @@ void Application::allTorrentsFinished()
         else if (shutdown)
             action = ShutdownAction::Shutdown;
 
-        if (!ShutdownConfirmDlg::askForConfirmation(action)) return;
+        if (will_exit_only) {
+            if (!pref->dontConfirmAutoExit()) {
+                bool exitConfirmed = false;
+                bool neverAskForExitConfirmationAgain = false;
+                ShutdownConfirmDlg::askForConfirmation(action, &exitConfirmed, &neverAskForExitConfirmationAgain);
```

Comment:
PS: Of course `neverAskForExitConfirmationAgain` will not be a pointer in this case.